build.cake

```diff
@@ -0,0 +1,81 @@
+// Arguments.
+var target = Argument("target", "Default");
+var configuration = Argument("configuration", "Release");
+
+// Define directories.
+var solutionFile = GetFiles("./Okta.Xamarin/*.sln").First();
+
+// Android and iOS.
+var androidProject = GetFiles("./Okta.Xamarin/Okta.Xamarin.Android/*.csproj").First();
+var iOSProject = GetFiles("./Okta.Xamarin/Okta.Xamarin.iOS/*.csproj").First();
+
+// Output folders.
+var artifactsDirectory = Directory(System.IO.Path.Combine(Environment.CurrentDirectory, "artifacts"));
+var androidOutputDirectory = Directory(System.IO.Path.Combine(artifactsDirectory, "Android"));
+var iOSOutputDirectory = Directory(System.IO.Path.Combine(artifactsDirectory, "iOS"));
+
+// Tests.
+var testsProject = GetFiles("./Okta.Xamarin/Okta.Xamarin.Test/*.csproj").First();
+
+Task("Clean")
+	.Does(() => 
+	{
+		CleanDirectory(artifactsDirectory);
+
+		DotNetBuild(solutionFile, settings => settings
+			.SetConfiguration(configuration)
+			.WithTarget("Clean")
+			.SetVerbosity(Verbosity.Minimal));
+	});
+
+Task("Restore-Packages")
+	.Does(() => 
+	{
+		NuGetRestore(solutionFile);
+	});
+
+Task("Build-Android")
+	.IsDependentOn("Restore-Packages")
+	.Does(() =>
+	{ 		
+		DotNetBuild(androidProject, settings =>
+			settings
+				.SetConfiguration(configuration)  
+				.WithProperty("OutputPath", androidOutputDirectory)         
+				.WithProperty("DebugSymbols", "false")
+				.WithProperty("TreatWarningsAsErrors", "false")
+				.SetVerbosity(Verbosity.Minimal));
+    });
+
+Task("Build-iOS")
+	.IsDependentOn("Restore-Packages")
+	.Does (() =>
+	{
+		DotNetBuild(iOSProject, settings => 
+			settings
+				.SetConfiguration(configuration)   
+				.WithTarget("Build")
+				.WithProperty("Platform", "iPhoneSimulator")
+				.WithProperty("OutputPath", iOSOutputDirectory)
+				.WithProperty("TreatWarningsAsErrors", "false")
+				.SetVerbosity(Verbosity.Minimal));
+	});
+
+Task("Run-Tests")
```

Comment:
Shouldn't tests tasks depend on build?

Reply:
My local experimentation implies that "DotNetCoreTest" builds prior to running the tests defined in the project.  Dependent projects are also built.  I tested this with the following:

```
./build.sh --target=Clean
./build.sh --target=Run-Tests
```

The tests ran and passed without issue.

Reply:
Oh yes, you are right. If you want to use the Cake build task instead, you can configure `DotNetCoreTest` to not build the project by doing this:

```
     DotNetCoreTest(string.Format("./src/{0}/{0}.csproj", name), new DotNetCoreTestSettings()
                {
                    Configuration = configuration,
                    NoBuild = true
                });
```

If you still want `DotNetCoreTest` to do that for you, I suggest adding a comment clarifying this.